docs/blueprints/EP031.rst

```diff
@@ -0,0 +1,273 @@
+
+      - table 2
+
+        - priority: Any
+        - match: <in_port=10,vlan_vid=20>  # Just in_port and vlan_vid
+        - instructions:
+
+          - action: pop_int
+          - action: <all original actions (set_queue, output, push/pop/swap vlan, etc.)>
+
+  - The choice between adding telemetry or not at the INT Sink Switch will be discussed in Section III.
+  - There are other steps for the INT Sink to be discussed later in Section III.
+
+
+3. NoviFlow switches in the path (a.k.a. INT Hop switch) will only need to add telemetry data to IPv4/TCP/UDP packets.
+
+   1. Example:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port=10,vlan_vid=20,eth_type=0x800,ip_proto=6>  # TCP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action: <all original actions (set_queue, output, push/pop/swap vlan, etc.)>
+
+        - priority: Higher (40000)
+        - match: <in_port=10,vlan_vid=20,eth_type=0x800,ip_proto=17>. # UDP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action: <all original actions (set_queue, output, push/pop/swap vlan, etc.)>
+
+
+
+III. Adding INT metadata at the INT Sink switch
+===============================================
+The NoviWare's INT implementation requires `send_report` action to be executed in Table 0. `send_report` is executed with higher priority than other INT actions, which means adding INT metadata at the INT Sink has to be performed before `send_report`. To address these requirements, the packets have to be re-injected into the pipeline using external connections via physical loops.
+
+To illustrate the challenge, consider an EVC terminating on INT Hop Z on port 23. The user packet with INT metadata comes from port 11. **mef_eline** would create the following flows (for simplicity, just one direction is presented):
+
+  0. **met_eline** default behavior:
+
+    - match:
+
+      - priority: DEFAULT (32678)
+      - match: <in_port=11,vlan_vid=20>
+
+    - instruction:
+
+      - action [set_queue, pop_vlan, etc.]
+      - action: output to port 23.
+
+To enable INT, first a physical loop has to be deployed. For this example, on INT Hop Z, port 1 is connected to port 2 by a physical fiber patch cord (done during commissioning). Then, the following flows need to be ADDED to the pipeline:
+
+  1. Adding INT metadata:
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=6>  # TCP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=17>. # UDP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
```

Comment:
@jab1982, regarding ` 3. There will be no concerns if INT Sink switches have loops for INT reports (Section III)`, I wonder if these loops should be configurable just so when pushing the new flows then given a dpid this NApp could derive which physical loops are expected to be used, how are you envisioning this part?